### ISSUE TITLE
[FEAT] Requests Timeout Definition

### ISSUE THREAD

**Issue by jplacht** [jplacht/fio_wrapper#12]
**Is your feature request related to a problem? Please describe.**

FIO Wrapper currently does not set a timeout for requests towards FIO in case FIO is not responding:

```python
# fio_adapter.py :: _do()

response = requests.request(
    method=http_method,
    url=endpoint,
    verify=self._ssl_verify,
    headers=self.headers(),
    params=params,
    json=data,
)
```

We should set one / allow setting one as FIO could be unavailable and the wrapper should not block all executions.

**Describe the solution you'd like**

1. setting a fixed timeout in the Adapter
2. allowing users to overwrite the standard timeout 

**Additional context**

Python `requests` can set the `timeout` parameter on individual requests, however a [Note](https://requests.readthedocs.io/en/stable/user/quickstart/#timeouts) mentions that this timeout is only related to the request receiving an response:

> timeout is not a time limit on the entire response download; rather, an exception is raised if the server has not issued a response for timeout seconds (more precisely, if no bytes have been received on the underlying socket for timeout seconds). If no timeout is specified explicitly, requests do not time out.

**Open points**

- Would it be required to set a timeout on a per-request level? E.g. setting a shorter timeout on Materials while Storage or GroupHub is able to wait longer on a FIO response?

**Comment by jplacht** [comment]
@rddaz2013 as you made a Fork I'm asking you directly: I guess a timeout is required, but would you have the need to set this individually or is a single overall timeout sufficient?

**Comment by jplacht** [comment]
> Yes should be added if it is clear which calls require an individual timeout. Maybe record which requests take a long time.

That would lead to an optional timeout variable on all requests, so e.g.,

```python
# current implementation
materials = fio.Material.get("DW")

# would also be callable as
materials = fio.Material.get("DW", timeout=2)
```

That would also be your preferred solution, right?

> In order not to overload the server, perhaps one option would be to save repetitive queries - similar to how https://oxylabs.io/blog/python-cache-how-to-use-effectively does

This is pretty neat; never stumpled upon this. Did you ever use it?

> _ps. hope your API key in Example.py is read-only https://blog.netwrix.com/2022/11/14/how-to-hide-api-keys-github/_

Hehe, it's already deleted.

**Comment by jplacht** [comment]
Merged to `master` with #14
